klaus/highlighting.py

```diff
@@ -34,17 +36,18 @@ def _format_lines(self, tokensource):
 
     def _lookup_ctag(self, token):
         matches = list(self._get_all_ctags_matches(token))
-        best_matches = self.get_best_ctags_matches(matches)
-        if not best_matches:
+        try:
+            best_match = next(self.get_best_ctags_matches(matches))
```

Comment:
I have not benchmarked it, I just noticed that you use `yield` so I assumed that it was to improve performance and not store everything in memory before returning from the function. Now that I think about it, using `yield` also makes the code nicer so maybe I just misinterpreted your choice.
I'm also happy with calling `list` if you think it is more clear. I use `next()` often so that may be why it does not seem unclear to me.
Anyway, I re-pushed with the list option, it probably makes little difference in the end :D